client/src/components/Operations.js

```diff
@@ -0,0 +1,157 @@
+import React, { Component } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { Paper, Grid } from '@material-ui/core/';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Divider from '@material-ui/core/Divider';
+import Typography from '@material-ui/core/Typography';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import Input from '@material-ui/core/Input';
+import { Link } from 'react-router-dom'
+
+const styles = theme => ({
+    rooti: {
+        ...theme.mixins.gutters(),
+        paddingTop: theme.spacing.unit * 4,
+        paddingBottom: theme.spacing.unit * 4,
+    },
+    withdraowButton: {
+        margin: theme.spacing.unit * 2,
+        backgroundColor: 'red'
+    },
+    depositButton: {
+        margin: theme.spacing.unit * 2,
+    },
+    textField: {
+        marginLeft: theme.spacing.unit,
+        marginRight: theme.spacing.unit,
+        width: 200
+    },
+    formControl: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        margin: theme.spacing.unit,
+        minWidth: 120,
+    },
+    selectEmpty: {
+        marginTop: theme.spacing.unit * 2,
+    },
+});
+
+class Operations extends Component {
+    constructor() {
+        super()
+        this.state = {
+            amount: '',
+            vendor: "",
+            category: ""
+        }
+    }
+
+    setTranscation = event => {
+        const value = event.target.value
+        const name = event.target.name
+        this.setState({ [name]: value })
+    }
+
+    deposit = () => {
+        const transcation = { amount: 0, vendor: this.state.vendor, category: this.state.category }
+        const amount = parseInt(this.state.amount)
+        transcation.amount = amount
+        if (transcation.amount && transcation.category && transcation.vendor) {
+            this.props.setNewTransaction(transcation)
+            this.setState({
+                amount: '',
+                vendor: "",
+                category: ""
+            }, () => { return true })
+        }
+        else {
+            return false
+        }
+    }
+
+    withdraw = () => {
```

Comment:
withdraw and deposit methods are almost the same -> you could have made them one smarter function...